doc/Custom-layouts.md

```diff
@@ -1,112 +1,114 @@
+* `key1` through `key8` is an older way to achieve the same effects. The directions are ordered as follows:
 
 key1 | key7 | key2
-:--: | :--: | :--:
+:-: | :--: | :-:
 key5 | key0 | key6
 key3 | key8 | key4
 
-The following properties are optionally supported:
+You can define a swipe only once with either compass-point or numeric notation. Unexpected Keyboard automatically puts a small legend in that direction from the center of the key.
+
+* `slider`: If `slider="true"`, and the key also has `w` and `e` properties, then the key tracks horizontal finger motion precisely and sends the `w` and `e` keystrokes repeatedly. In built-in layouts, this makes the space bar send left and right characters as the user slides on the space bar.
+* `anticircle`: The key value to send when doing an anti-clockwise gesture on the key. (The clockwise gesture applies a Shift modifier and the round-trip gesture applies a Fn modifier. These results can be redefined using a modmap, as explained below.)
+
+### Layout
+A key may have the following properties to control the row's layout:
+* `width`: The width of the key, a positive floating-point value. It defaults to 1.0
+* `shift`: How much empty space to add to the left of this key, a non-negative floating-point value. It defaults to 0.0
+
+Normally, a key's width is 1.0 unit. Unexpected Keyboard occupies the full width of the screen, and the row defining the highest number of units (in widths plus shifts) is as wide as the screen. A row whose width is a smaller number of units has empty space on the right.
 
-* `width`: The width of the key relative to the normal width. Defaults to `1` and accepts a positive floating point value.
-* `shift`: How much empty space to add to the left of this key. Defaults to `0` and accepts a non-negative floating point value.
-* `indication`: An extra label to show under the main label, intended to be used as a legend for 2A typing (e.g. `<key key0="2" indication="ABC" />`). Caution: if you have `key8` defined, it overlaps!
-* `slider`: If set to `true`, the keys `w` and `e` are sent repeatedly when the key is being slid on. Intended to be used on the space bar, and in fact used on the default space bar.
-* `anticircle`: The key value to send when doing an anti-clockwise circle gesture on the key. The clockwise circle and round-trip gestures are not configurable that way.
+### Extra legend
+* `indication`: An optional extra legend to show under the main label. For example, `<key key0="2" indication="ABC" />` displays ABC at the bottom of the 2 key, as on a pinpad or some telephones. If the key also defines a downward swipe with `s` or `key8`, the legends overlap.
 
 ## Possible key values
+The properties that define the effects of tapping or swiping a key may be one of the built-in strings documented on [this page](Possible-key-values). For example, `se="cut"` says that a southeast swipe produces the "cut" character (Ctrl-C).
```

Comment:
I'd rather not use "character" for talking of anything else than a piece of text containing only one visual character. Internally, the "cut" key is an "editing key" but I'd prefer not to over explain these internal details. "Key" is fine.
```suggestion
The properties that define the effects of tapping or swiping a key may be one of the built-in strings documented on [this page](Possible-key-values). For example, `se="cut"` says that a southeast swipe produces the "cut" key (Ctrl-C).
```

Reply:
Okay. Ctrl-C is a control "character" (or "code") but I agree, reader wants to deal with keys, not bytes.